src/org/openlcb/swing/NodeSelector.java

```diff
@@ -28,18 +28,31 @@
  * from a MimicNodeStore
  *
  * @author	Bob Jacobsen   Copyright (C) 2012
- * @version	$Revision$
  */
 public class NodeSelector extends JPanel  {
-    /** Comment for <code>serialVersionUID</code>. */
-    private static final long serialVersionUID = 1714640844679691939L;
-    
+
     private final PropertyChangeListener propertyChangeListener;
     MimicNodeStore store;
     JComboBox<ModelEntry> box;
     private DefaultComboBoxModel<ModelEntry> model = new DefaultComboBoxModel<ModelEntry>();
     private boolean seenLight = false;
+    private int termCount = 2; // how many terms to keep in ID string
+
+
+    /**
+     * Constructor that allows you to set the length of the displayed ID
+     * @param store Node store containing the existing network
+     * @param termCount Number of ID terms to include in the displayed ID
```

Comment:
`// This is how many properties will be shown, in the order of User Name, User Description, Manufacturer+Model, Software version. Only non-empty values count.`